File: chore-manager-frontend/src/app/model/dto/task.ts
import {BaseDto} from "./base-dto";
import {Item} from "./item";
import {OccurrenceInfo} from "../occurrence";
import {ReminderInfo} from "../reminder";
import {CompletionHistoryItem} from "../completion-history-item";
import {Recurrence} from "../recurrence";
import {DeferInfo} from "../defer";

export interface Task extends BaseDto {
  description: string;
  dateCreated?: string;
  item: Item;
  recurrence: Recurrence;
  occurrence: OccurrenceInfo;
  defer?: DeferInfo;
  reminder: ReminderInfo;
  history: CompletionHistoryItem[];
  enabled: boolean;
}

File: chore-manager-backend/chore-manager-core/src/main/java/net/lazars/chores/core/model/Task.java
package net.lazars.chores.core.model;

import java.time.LocalDate;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import net.lazars.chores.core.model.recurrence.Recurrence;

@Data
@EqualsAndHashCode(callSuper = true)
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class Task extends BaseModel {

  private String description;

  private LocalDate dateCreated;

  private Item item;

  private Recurrence recurrence;

  private OccurrenceInfo occurrence;

  private DeferInfo defer;

  private List<CompletionHistoryItem> history;

  private ReminderInfo reminder;

  private boolean enabled;
}
